tests/test_appclass.py

```diff
@@ -0,0 +1,159 @@
+import aiida
+import pytest
+from packaging import version
+
+from aiidalab.app import AiidaLabApp
+
+
+def test_init_refresh(generate_app):
+    app = generate_app()
+    assert len(app.available_versions) == 0
+    # After refresh the availale_versions traitslet is updated
+    app.refresh()
+    assert len(app.available_versions) != 0
+
+
+def test_prereleases(generate_app):
+    app = generate_app()
+
+    # without prereleases tick
+    app.refresh()
+    assert app.has_prereleases
+    assert app.include_prereleases is False
+
+    # tick prereleases tick
+    app.include_prereleases = True
+    assert "v23.01.0b1" in app.available_versions
+
+
+@pytest.mark.skipif(
+    version.parse(aiida.__version__).major != 2, reason="only pass for aiida 2.x"
+)
+def test_dependencies(generate_app):
+    app: AiidaLabApp = generate_app()
+    app.refresh()
+
+    # The version `v22.11.0` is incompatible while `v22.11.1` is compatible
+    app.version_to_install = "v22.11.0"
+    assert app.strict_dependencies_validation is False
+
+    app.version_to_install = "v22.11.1"
+    assert app.strict_dependencies_validation is True
+
+
+# The data for test purpose as fixture
+@pytest.fixture
+def generate_app():
+    """Fixture to construct a new AiiDALabApp instance for testing."""
+
+    def _generate_app(
+        name="quantum-espresso",
+        aiidalab_apps_path="/home/jovyan/apps",
+        app_data=None,
+        watch=False,
+    ):
+        if app_data is None:
+            app_data = EXAMPLE_APP_REGISTRY_ENTRY
+
+        app = AiidaLabApp(name, app_data, aiidalab_apps_path, watch=False)
+
+        return app
+
+    return _generate_app
+
+
+EXAMPLE_APP_REGISTRY_ENTRY = {
```

Comment:
Please put this in a separate file (maybe in `static` directory?). It would also look a bit nice in YAML format, but that's up to you, I don't have a strong opinion.

Reply:
done in 2ed9d6f